.github/workflows/ci.yml

```diff
@@ -45,6 +45,52 @@ jobs:
         run: |
           make -C build/ help | grep demo | tr -d '. ' | xargs make -C build/
           make -C demos/jobs/jobs_demo_mosquitto
+  build-check-demos-cpp:
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Clone This Repo
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: Rename source files
+        run: |
+          for demo in mqtt http shadow defender
```

Comment:
Instead of using include filter, an exclude filter approach can be used to reduce the nested for loop into a single command as
`find ./demos -name '*.c' -not -path "*ota*" -not -path "*pkcs11*" -type f -exec sh -c 'for f; do mv -- $f ${f%.*}.cpp; done' _ {} +`

Reply:
I didn't think a nested loop was any worse than one super long command, and probably easier to read as a loop. One difference is that the command above changes every .c file, including ones in helper directories on a different directory level, while the loop only changes the files that are in the same directory as each demo. Regardless, I made the proper CMake changes to allow the above command to work, and left it as a suggestion in another comment if you decide to commit it

Reply:
To mimic the same behavior as the nested for loop, the command could use an include filter as
`find ./demos -name '*.c'  -path "demos/mqtt*" -o -path "demos/http*" -o -path "demos/shadow*" -o -path "demos/defender*" -type f -exec sh -c 'for f; do mv -- $f ${f%.*}.cpp; done' _ {} +`

It's a minor style preference I have to limit the number of lines for bash logic in the workflow file. Either way, I'm fine with whichever logic is chosen. It may be beneficial to rename the job's step name to "Rename demo source files to .cpp extension" to make it more understandable.